app/src/main/java/com/island/island/Activities/EditProfileActivity.java

```diff
@@ -10,72 +11,115 @@
+    private Uri headerImageUri = null;
 
     private static final int SELECT_PROFILE_IMAGE = 1;
     private static final int SELECT_HEADER_IMAGE = 2;
 
     @Override
-    protected void onCreate(Bundle savedInstanceState)
-    {
+    protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_edit_profile);
         Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
 
-        Profile profile = ProfileDatabase.getInstance(getApplicationContext()).get(Utils.getUser(this));
+        Profile profile = ProfileDatabase.getInstance(
+                getApplicationContext()).get(Utils.getUser(this));
 
         TextView userName = (TextView) findViewById(R.id.profile_user_name);
+        profileImage = (ImageView) findViewById(R.id.profile_profile_image);
+        headerImage = (ImageView) findViewById(R.id.profile_header_image);
         aboutMe = (EditText) findViewById(R.id.edit_profile_about_me);
 
+        prevProfileImageUri = profile.getProfileImageUri();
+        prevHeaderImageUri = profile.getHeaderImageUri();
+
         userName.setText(profile.getUsername());
         aboutMe.setText(profile.getAboutMe());
+
+        ImageUtils.setProfileImageSampled(getApplicationContext(), profileImage,
+                profile.getProfileImageUri());
+        ImageUtils.setHeaderImageSampled(getApplicationContext(), headerImage,
+                profile.getHeaderImageUri());
     }
 
     @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data)
-    {
-        if (resultCode == RESULT_OK)
-        {
-            if (requestCode == SELECT_PROFILE_IMAGE)
-            {
-                // TODO: set profile image and save uri to database
-                Uri selectedImageUri = data.getData();
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (resultCode == RESULT_OK) {
+            if (requestCode == SELECT_PROFILE_IMAGE) {
+                profileImageUri = data.getData();
+                ImageUtils.setProfileImageSampled(getApplicationContext(),
+                        profileImage, profileImageUri);
+
             }
-            else if(requestCode == SELECT_HEADER_IMAGE)
-            {
-                // TODO: set header image
-                Uri selectedImageUri = data.getData();
+            else if(requestCode == SELECT_HEADER_IMAGE) {
+                headerImageUri = data.getData();
+                ImageUtils.setHeaderImageSampled(getApplicationContext(),
+                        headerImage, headerImageUri);
             }
         }
     }
 
-    public void saveProfile(View view)
-    {
-        // TODO: handle image stuff for saving
+    public void saveProfile(View view) {
         String newAboutMeText = aboutMe.getText().toString();
         String myUsername = PreferenceManager.getDefaultSharedPreferences(getApplicationContext())
```

Comment:
I think I wrote this, but since we are updating this method, we should probably use the Utils.getUser() method here.